controllers/dockermachine_controller.go

```diff
@@ -47,17 +68,257 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 
 	dockerMachine := &infrastructurev1alpha2.DockerMachine{}
 	if err := r.Client.Get(ctx, req.NamespacedName, dockerMachine); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
 		log.Error(err, "failed to get dockerMachine")
+		return ctrl.Result{}, err
+	}
+
+	// Find the owner reference
+	var machineRef *metav1.OwnerReference
+	for _, ref := range dockerMachine.OwnerReferences {
+		if ref.Kind == machineKind.Kind && ref.APIVersion == machineKind.Version {
+			machineRef = &ref
+			break
+		}
+	}
+	if machineRef == nil {
+		log.Info("did not find matching machine reference")
+		return ctrl.Result{}, nil
+	}
+
+	// Get the cluster api machine
+	machine := &capiv1alpha2.Machine{}
+	machineKey := client.ObjectKey{
+		Namespace: req.Namespace,
+		Name:      machineRef.Name,
+	}
+
+	if err := r.Get(ctx, machineKey, machine); err != nil {
```

Comment:
this will fail on rbac, you'll need to add this line:
`// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;machines,verbs=get;list;watch` to this group and regen manifests https://github.com/kubernetes-sigs/cluster-api-provider-docker/pull/148/files#diff-5a0ef128e5d699a37a70d38f9a51d847R61-R62